libminifi/test/aws-tests/AWSCredentialsServiceTest.cpp

```diff
@@ -65,3 +77,28 @@ TEST_CASE_METHOD(AWSCredentialsServiceTestAccessor, "Test expired credentials ar
   // Check for credential refresh
   REQUIRE_FALSE(aws_credentials_impl->getAWSCredentials()->IsExpired());
 }
+
+TEST_CASE_METHOD(AWSCredentialsServiceTestAccessor, "Test credentials from default credential chain are always refreshed", "[credentialRefresh]") {
+  setEnvironmentCredentials("key", "secret");
+  plan->setProperty(aws_credentials_service, "Use Default Credentials", "true");
+  aws_credentials_service->enable();
+  assert(aws_credentials_service->getControllerServiceImplementation() != nullptr);
+  auto aws_credentials_impl = std::static_pointer_cast<minifi::aws::controllers::AWSCredentialsService>(aws_credentials_service->getControllerServiceImplementation());
+
+  // Check intial credentials
+  REQUIRE(aws_credentials_impl->getAWSCredentials());
+  REQUIRE(aws_credentials_impl->getAWSCredentials()->GetAWSAccessKeyId() == "key");
+  REQUIRE(aws_credentials_impl->getAWSCredentials()->GetAWSSecretKey() == "secret");
+  REQUIRE_FALSE(aws_credentials_impl->getAWSCredentials()->IsExpired());
+
+  // Set new credentials
+  setEnvironmentCredentials("key2", "secret2");
+
+  // Check for credential refresh
+  REQUIRE(aws_credentials_impl->getAWSCredentials());
+  REQUIRE(aws_credentials_impl->getAWSCredentials()->GetAWSAccessKeyId() == "key2");
+  REQUIRE(aws_credentials_impl->getAWSCredentials()->GetAWSSecretKey() == "secret2");
+  REQUIRE_FALSE(aws_credentials_impl->getAWSCredentials()->IsExpired());
+}
+
+}
```

Comment:
```suggestion
}  // namespace
```

Reply:
Fixed in c1a18bb5c2c60662e095ff43b63b2b0e53bd8635